django_tasks/backends/database/models.py

```diff
@@ -226,3 +227,13 @@ def set_failed(self, exc: BaseException) -> None:
                 "traceback",
             ]
         )
+
+    def duplicate(self) -> Self:
```

Comment:
Suggestion: This API feels like one better suited for the `TaskResult`, rather than being backend specific.